Formula/samba.rb

```diff
@@ -0,0 +1,133 @@
+    ENV.append "CFLAGS", "-include #{buildpath}/lib/util/debug.h"
+    system "./configure",
+           "--disable-cephfs",
+           "--disable-cups",
+           "--disable-iprint",
+           "--disable-glusterfs",
+           "--disable-python",
+           "--without-acl-support",
+           "--without-ad-dc",
+           "--without-ads",
+           "--without-dnsupdate",
+           "--without-ldap",
+           "--without-libarchive",
+           "--without-json",
+           "--without-ntvfs-fileserver",
+           "--without-pam",
+           "--without-regedit",
+           "--without-syslog",
+           "--without-utmp",
+           "--without-winbind",
+           "--prefix=#{prefix}"
+    system "make"
+    system "make", "install"
+    on_macos do
+      # macOS has its own SMB daemon as /usr/sbin/smbd, so rename our smbd to samba-dot-org-smbd to avoid conflict.
+      # samba-dot-org-smbd is used by qemu.rb .
+      mv "#{sbin}/smbd", "#{sbin}/samba-dot-org-smbd"
+    end
+  end
+
+  def caveats
+    on_macos do
+      <<~EOS
+        macOS has its own SMB daemon as /usr/sbin/smbd, so Samba version of smbd is installed as "#{HOMEBREW_PREFIX}/sbin/samba-dot-org-smbd"
+
+        On macOS, Samba should be executed as a non-root user: https://bugzilla.samba.org/show_bug.cgi?id=8773
+      EOS
+    end
+  end
+
+  test do
+    on_macos do
+      system "#{sbin}/samba-dot-org-smbd", "--version"
+      system "#{sbin}/samba-dot-org-smbd", "--help"
+    end
+
+    on_linux do
+      system "#{sbin}/smbd", "--version"
+      system "#{sbin}/smbd", "--help"
+    end
+
+    port = free_port
+
+    mkdir_p "samba/state"
+    mkdir_p "samba/data"
+    (testpath/"samba/data/hello").write "hello"
+    # mimic smb.conf generated by qemu
+    # https://github.com/qemu/qemu/blob/v6.0.0/net/slirp.c#L862
+    (testpath/"smb.conf").write <<~EOS
+      [global]
+      private dir=#{testpath}/samba/state
+      interfaces=127.0.0.1
+      bind interfaces only=yes
+      pid directory=#{testpath}/samba/state
+      lock directory=#{testpath}/samba/state
+      state directory=#{testpath}/samba/state
+      cache directory=#{testpath}/samba/state
+      ncalrpc dir=#{testpath}/samba/state/ncalrpc
+      log file=#{testpath}/samba/state/log.smbd
+      smb passwd file=#{testpath}/samba/state/smbpasswd
+      security = user
+      map to guest = Bad User
+      load printers = no
+      printing = bsd
+      disable spoolss = yes
+      usershare max shares = 0
+      [test]
+      path=#{testpath}/samba/data
+      read only=no
+      guest ok=yes
+      force user=#{ENV["USER"]}
+    EOS
+
+    pid = 0
+    on_macos do
+      pid = fork { exec sbin/"samba-dot-org-smbd", "-F", "--configfile=smb.conf", "--port=#{port}" }
+    end
+    on_linux do
+      pid = fork { exec sbin/"smbd", "-F", "--configfile=smb.conf", "--port=#{port}" }
+    end
```

Comment:
I believe this needs 
```suggestion
    on_macos do
      pid = fork { exec sbin/"samba-dot-org-smbd", "-S", "-F", "--configfile=smb.conf", "--port=#{port}" }
    end
    on_linux do
      pid = fork { exec sbin/"smbd", "-S", "-F", "--configfile=smb.conf", "--port=#{port}" }
    end
```

Reply:
Added